src/server/v2.0/handler/label.go

```diff
@@ -0,0 +1,218 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/common/rbac/system"
+	"github.com/goharbor/harbor/src/controller/project"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/label"
+	pkg_model "github.com/goharbor/harbor/src/pkg/label/model"
+	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/label"
+	"strings"
+)
+
+func newLabelAPI() *labelAPI {
+	return &labelAPI{
+		labelMgr:   label.Mgr,
+		projectCtl: project.Ctl,
+	}
+}
+
+type labelAPI struct {
+	BaseAPI
+	labelMgr   label.Manager
+	projectCtl project.Controller
+}
+
+func (lAPI *labelAPI) CreateLabel(ctx context.Context, params operation.CreateLabelParams) middleware.Responder {
+	label := &pkg_model.Label{}
+	lib.JSONCopy(label, params.Label)
+
+	label.Level = common.LabelLevelUser
+	switch label.Scope {
+	case common.LabelScopeGlobal:
+		label.ProjectID = 0
+	case common.LabelScopeProject:
+		exist, err := lAPI.projectCtl.Exists(ctx, label.ProjectID)
+		if err != nil {
+			return lAPI.SendError(ctx, errors.Wrapf(err, "failed to check the existence of project %d", label.ProjectID))
+		}
+		if !exist {
+			return lAPI.SendError(ctx, errors.New(nil).WithMessage("project %d not found", label.ProjectID).WithCode(errors.BadRequestCode))
+		}
```

Comment:
The project will be checked in the `requireAccess` method, do we need these checking here?

Reply:
remove the dup check